### PR TITLE
Stats: Add Mobile Apps promo card via DotPager UI, take 3

### PR DESCRIPTION
**Pull request by a8ck3n**
#### Proposed Changes

Adds the Jetpack mobile apps promo card alongside the Yoast SEO one using the `DotPager` component. The Yoast card is still the default.

Before:

APromoSEO

After:

NewPromoCards

#### Testing Instructions

1. Visit the Calypso Live link.
2. Visit the Stats page.
3. Scroll down and confirm the `DotPager` card is present.
4. Confirm the Yoast SEO card is the default. 
5. Confirm the mobile apps card is available on the 2nd page.
6. Confirm it behaves appropriately on smaller screen sizes.
7. Test on a self-hosted Jetpack site and confirm the Yoast card and the `DotPager` UI are not used. Only the Jetpack mobile apps CTA should be shown.

#### Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #69230 and supersedes #70463.

Currently just updating the Stats page. The designs call for the same promo card on the Insights and Ads pages but it looks overpowering there with the older page layout. I'm thinking we should wait until we "grid" those layouts before adding this component in.

### CODE REVIEW

File: client/my-sites/stats/site.jsx [hunk 1]
```diff
@@ -368,20 +374,36 @@ class StatsSite extends Component {
...
-							href={ `/plugins/wordpress-seo-premium/${ slug }` }
-						/>
+				{ ! isOdysseyStats && isJetpackNonAtomic && (
+					<div className="stats__promo-container">
+						<div className="stats__promo-card">
+							<MobilePromoCard className="stats__promo-card-apps" />
+						</div>
+					</div>
+				) }
+				{ ! isJetpackNonAtomic && (
```
Comment (jsnmoon) [inline client/my-sites/stats/site.jsx:384]:
```suggestion
				{ /** Yoast promotion is disabled in Odyssey, which avoids an API call to `plugins`. */ }
				{ ! isOdysseyStats && ! isJetpackNonAtomic && (
```

No need to promote Yoast in Odyssey Stats.

File: client/my-sites/stats/site.jsx [hunk 2]
```diff
@@ -368,20 +374,36 @@ class StatsSite extends Component {
...
-							clickEvent="calypso_stats_wordpress_seo_premium_banner_click"
-							headerText={ translate( 'Increase site visitors with Yoast SEO Premium' ) }
-							contentText={ translate(
-								'Purchase Yoast SEO Premium to ensure that more people find your incredible content.'
-							) }
-							ctaText={ translate( 'Learn more' ) }
-							image={ wordpressSeoIllustration }
-							href={ `/plugins/wordpress-seo-premium/${ slug }` }
-						/>
+				{ ! isOdysseyStats && isJetpackNonAtomic && (
```
Comment (jsnmoon) [inline client/my-sites/stats/site.jsx:377]:
```suggestion
				{ isJetpackNonAtomic && (
```

Promoting the Jetpack mobile app makes sense in Odyssey Stats.

File: client/my-sites/stats/site.jsx [hunk 3]
```diff
@@ -368,20 +374,36 @@ class StatsSite extends Component {
 					</div>
 				</div>
 				{ /** Promo Card is disabled for Odyssey because it doesn't make much sense in the context, which also removes an API call to `plugins`. */ }
```
Comment (jsnmoon) [inline client/my-sites/stats/site.jsx:376]:
```suggestion
```

File: client/my-sites/stats/site.jsx [hunk 4]
```diff
@@ -189,7 +192,10 @@ class StatsSite extends Component {
 	}
 
 	renderStats() {
-		const { date, siteId, slug, isJetpack, isSitePrivate, isOdysseyStats } = this.props;
+		const { date, siteId, slug, isAtomic, isJetpack, isSitePrivate, isOdysseyStats } = this.props;
+
+		// Yoast promo card is only shown on Atomic sites.
+		const isJetpackNonAtomic = isJetpack && ! isAtomic;
```
Comment (jsnmoon) [inline client/my-sites/stats/site.jsx:198]: This logic is hard to parse. If we only want to show the Yoast promo for Atomic sites, why do we create a variable checking for a non-atomic Jetpack site?

Something like this would be easier to understand:

```js
		// Yoast promo card is only shown on Atomic sites.
		const shouldPromoteYoast = isAtomic;
```

or

```js
		// Non-atomic Jetpack sites should not see Yoast promotions.
		const shouldHideYoastPromo = isJetpack && ! isAtomic;
```

We can also include `isOdysseyStats` directly into one of those conditionals that control Yoast promo visibility.

File: client/my-sites/stats/site.jsx [hunk 5]
```diff
@@ -368,20 +374,36 @@ class StatsSite extends Component {
...
+											'Purchase Yoast SEO Premium to ensure that more people find your incredible content.'
+										) }
+										ctaText={ translate( 'Learn more' ) }
+										image={ wordpressSeoIllustration }
+										href={ `/plugins/wordpress-seo-premium/${ slug }` }
+									/>
+								</div>
+								<div>
+									<MobilePromoCard className="stats__promo-card-apps" />
+								</div>
```
Comment (jsnmoon) [inline client/my-sites/stats/site.jsx:404]:
```suggestion
								<PromoCardBlock
									productSlug="wordpress-seo-premium"
									impressionEvent="calypso_stats_wordpress_seo_premium_banner_view"
									clickEvent="calypso_stats_wordpress_seo_premium_banner_click"
									headerText={ translate( 'Increase site visitors with Yoast SEO Premium' ) }
									contentText={ translate(
										'Purchase Yoast SEO Premium to ensure that more people find your incredible content.'
									) }
									ctaText={ translate( 'Learn more' ) }
									image={ wordpressSeoIllustration }
									href={ `/plugins/wordpress-seo-premium/${ slug }` }
								/>
								<MobilePromoCard className="stats__promo-card-apps" />
```

No need for wrapping `div`s.

**Review by jsnmoon — COMMENTED** [review]
Looks good! I left some suggestions to improve code readability.

I also assumed this to be blocked until we add Tracks events to the MobilePromoCard component, so I applied the requisite label.

File: client/my-sites/stats/site.jsx (hunk 4 above)
Comment (a8ck3n) [reply client/my-sites/stats/site.jsx:198]: Yes, definitely. I'll clean this up a bit so the logic is more obvious.

**Comment by a8ck3n** [comment]
> Looks good! I left some suggestions to improve code readability.

Thanks for the review @jsnmoon. I've covered your feedback in #71201 which handles the new integration approach (with Tracks in the caller instead of in the component).
